source/jormungandr/tests/departure_board_tests.py

```diff
@@ -244,3 +244,38 @@ def test_departures(self):
 
         assert len(response["departures"][1]["stop_date_time"]["additional_informations"]) == 1
         assert response["departures"][1]["stop_date_time"]["additional_informations"][0] == "on_demand_transport"
+
+    def test_vj_comment(self):
+        """
+        test that a comment on a vehicle journey is linked to each stop_times in stop_schedules
+        but only in the header on route_schedules
+        """
+        # Check if the comment is on the stop_time
+        response = self.query_region("stop_points/stop1/stop_schedules?from_datetime=20120615T080000")
+        assert 'error' not in response
+        assert 'stop_schedules' in response
+        assert len(response["stop_schedules"]) == 1
+        assert len(response["notes"]) == 1
+        assert len(response["stop_schedules"][0]["date_times"]) == 4
+
+        for date_time in response["stop_schedules"][0]["date_times"]:
+            if len(date_time["links"]) > 1 and \
+               len([l for l in date_time["links"] if l["type"] == "vehicle_journey" and l["id"] == "all"]) == 1:
+                assert len([l for l in date_time["links"] if l["type"] == "notes" and l["id"] == response["notes"][0]["id"]]) == 1
+
+        # Check that the comment is only on the header
+        response = self.query_region("lines/line:A/route_schedules?from_datetime=20120615T080000")
+        assert 'error' not in response
+        assert 'route_schedules' in response
+        assert len(response["route_schedules"]) == 1
+        assert len(response["notes"]) == 1
+        assert len(response["route_schedules"][0]["table"]["headers"]) == 4
+
+        for header in response["route_schedules"][0]["table"]["headers"]:
+            if len([l for l in header["links"] if l["type"] == "vehicle_journey" and l["id"] == "all"]) == 1:
+                assert len([l for l in header["links"] if l["type"] == "notes" and l["id"] == response["notes"][0]["id"]]) == 1
+
+        for row in response["route_schedules"][0]["table"]["rows"]:
+            for date_time in row["date_times"]:
+                assert len([l for l in date_time["links"] if l["type"] == "notes" and l["id"] == response["notes"][0]["id"]]) == 0
```

Comment:
On all those tests, I think it would be more readable to declare, name and maybe comment the collections you test.
It would also shorten lines a bit :smile: